astropy/coordinates/representation.py

```diff
@@ -538,12 +773,24 @@ def _combine_operation(self, op, other, reverse=False):
             Whether the operands should be reversed (e.g., as we got here via
             ``self.__rsub__`` because ``self`` is a subclass of ``other``).
         """
+        self._raise_if_has_differentials(op.__name__)
+
         result = self.to_cartesian()._combine_operation(op, other, reverse)
         if result is NotImplemented:
             return NotImplemented
         else:
             return self.from_cartesian(result)
 
+    # We need to override this setter to support differentials
+    @BaseRepresentationOrDifferential.shape.setter
+    def shape(self, shape):
+        # See: https://stackoverflow.com/questions/3336767/ for an example
+        BaseRepresentationOrDifferential.shape.fset(self, shape)
```

Comment:
The problem occurs if, e.g., the representation has a proper (2,5) shape and the differential is really (5,) but broadcast to (2,5). Then you could not change the shape to (5, 2) or even just (10,):
```
In [14]: a=np.broadcast_to(np.ones(5), (2,5))

In [15]: a.shape = 10,
# AttributeError: incompatible shape for a non-contiguous array
```